### PR TITLE
Make health bars show rough percentage of max hp

### PR DESCRIPTION
**Pull request by LoneWolfHT**
From my quick testing this works fine, and fixes things like an hp_max of 10 and hp of 10 showing players at half health

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -50,7 +50,7 @@ minetest.register_entity("gauges:hp_bar", {
 			return
 		end
 
-		local hp = min(player:get_hp(), 20)
+		local hp = min(math.ceil((player:get_hp() / player:get_properties().hp_max) * 20), 20)
```
Comment (Calinou) [inline init.lua]: Should we also handle the case where `hp_max` is greater than 20? This would be a good opportunity to do it at the same time.

**Comment by SmallJoker** [comment]
FWIW, Minetest uses the following code to adjust the health bar to a fixed width (or max value): https://github.com/minetest/minetest/blob/master/builtin/game/statbars.lua#L30-L37

**Comment by LoneWolfHT** [comment]
> 
> 
> FWIW, Minetest uses the following code to adjust the health bar to a fixed width (or max value): https://github.com/minetest/minetest/blob/master/builtin/game/statbars.lua#L30-L37

Shall I just copy that over and attribute it with a code comment holding https://github.com/minetest/minetest/blob/6de8d77e17017cd5cc7b065d42566b6b1cd076cc/builtin/game/statbars.lua#L30-L37 ?

File: init.lua (hunk 1 above)
Comment (LoneWolfHT) [reply init.lua]: I think both this code and the new code will account for that

**Comment by LoneWolfHT** [comment]
Ready for review again, did what I asked about in my last comment

File: init.lua [hunk 2]
```diff
@@ -50,7 +50,11 @@ minetest.register_entity("gauges:hp_bar", {
 			return
 		end
 
-		local hp = min(player:get_hp(), 20)
+		-- hp calc credit: https://github.com/minetest/minetest/blob/6de8d77e17017cd5cc7b065d42566b6b1cd076cc/builtin/game/statbars.lua#L30-L37
+		local current = player:get_hp()
+		local max_display = math.max(20, math.max(player:get_properties().hp_max, current))
```
Comment (SmallJoker) [inline init.lua]: `20` is not necessary the minimal HP. Why not `math.max(player:get_properties().hp_max, current)` alone?

PS: The same applies for breath.

Comment (LoneWolfHT) [reply init.lua]: Missed that, should be good now

**Comment by SmallJoker** [comment]
```
2021-10-13 19:43:07: ERROR[Main]: generateImage(): Could not load image "health_3.3333333333333.png" while building texture; Creating a dummy image
2021-10-13 19:43:07: ERROR[Main]: generateImage(): Could not load image "breath_11.png" while building texture; Creating a dummy image
```

**Comment by LoneWolfHT** [comment]
> 
> 
> ```
> 2021-10-13 19:43:07: ERROR[Main]: generateImage(): Could not load image "health_3.3333333333333.png" while building texture; Creating a dummy image
> 2021-10-13 19:43:07: ERROR[Main]: generateImage(): Could not load image "breath_11.png" while building texture; Creating a dummy image
> ```

huh, the breath error is weird, should be fixed now

File: init.lua [hunk 3]
```diff
@@ -50,8 +60,8 @@ minetest.register_entity("gauges:hp_bar", {
 			return
 		end
 
-		local hp = min(player:get_hp(), 20)
-		local breath = min(player:get_breath(), max_breath)
+		local hp = scaleToDefault(player, "hp")
+		local breath = scaleToDefault(player, "breath")
 
 		if self.hp ~= hp or self.breath ~= breath then
```
Comment (SmallJoker) [inline init.lua:66]: Undefined `max_breath` below. Should be `breath + 1`  or `if breath > max.breath`

Comment (LoneWolfHT) [reply init.lua:66]: Sorry, should have tested this myself. Confirmed working, with higher and lower max hp/breath

**Comment by SmallJoker** [comment]
:+1: looks good now. Thanks.
I'll merge this in a few days unless Calinou has other plans.

**Pull request merged**